src/manifolds/EssentialManifold.jl

```diff
@@ -0,0 +1,508 @@
+````math
+f(t) = f_1 + f_2, \quad f_i = \frac{1}{2} θ^2_i(t), \quad θ_i(t)=d(R_{p_i},R_z(t)R_{b_i}) \text{ for } i=1,2,
+````
+where $d(⋅,⋅)$ is the distance function in $SO(3)$, is minimized. Further, the group $H_z$ acting
+on the left on $SO(3)^2$ is defined as
+````math
+H_z = \{(R_z(θ),R_z(θ))\colon θ \in [-π,π) \},
+````
+where $R_z(θ)$ is the rotation around the z axis with angle $θ$. Points in $H_z$ are denoted by
+$S_z$. Then, the logarithm is defined
+as
+````math
+\log_p (S_z(t_{\text{opt}})q) = [\text{Log}(R_{p_i}^T R_z(t_{\text{opt}})R_{b_i})]_{i=1,2},
+````
+where $\text{Log}$ is the [`logarithm`](@ref log(::Rotations, ::Any...)) on $SO(3)$. For more
+details see [^TronDaniilidis2017].
+"""
+function log!(M::EssentialManifold, X, p, q)
+    # compute the closest representative of q
+    t = 0
+    f_min = Inf
+    q2 = deepcopy(q)
+
+    if !M.is_signed
+        q2min = deepcopy(q)
+        for k in 1:4
+            #flip sign in q to get another member of its equivalence class
+            if k == 2
+                q2[1][2:3, :] = -q[1][2:3, :]
+                q2[2][[1 3], :] = -q[2][[1 3], :]
+            elseif k == 3
+                q2[2][1:2, :] = -q[2][1:2, :]
+            elseif k == 4
+                q2[1][2:3, :] = -q[1][2:3, :]
+                q2[2][2:3, :] = -q[2][2:3, :]
+            end
+            t_temp, f_temp = dist_min_angle_pair(p, q2)
+            if f_temp < f_min
+                q2min = deepcopy(q2)
+                f_min = f_temp
+                t = t_temp
+            end
+        end
+        q2 = q2min
+    else
+        t, _ = dist_min_angle_pair(p, q2)
+    end
+    Rz = [cos(t) -sin(t) 0; sin(t) cos(t) 0; 0 0 1]
+    representative_q = zeros(size(q2))
+    representative_q = [Rz * q2[1], Rz * q2[2]]
+
+    # use the logarithmic map in SO(3)^2 and the corresponding geodesic between p and representative_q to compute the solution
+    log!.(Ref(M.manifold), X, p, representative_q)
+    return X
+end
+
+@doc raw"""
+    dist_min_angle_pair(p, q)
+
+This function computes the global minimizer of the function
+````math
+f(t) = f_1 + f_2, \quad f_i = \frac{1}{2} θ^2_i(t), \quad θ_i(t)=d(R_{p_i},R_z(t)R_{b_i}) \text{ for } i=1,2,
+````
+for the given values. This is done by finding the discontinuity points $t_{d_i}, i=1,2$ of its derivative
+and using Newton's method to minimize the function over the intervals $[t_{d_1},t_{d_2}]$ and $[t_{d_2},t_{d_1}+2π]$
+separately. Then, the minimizer for which $f$ is minimal is chosen and given back together with the minimal value.
+For more details see Algorithm 1 in [^TronDaniilidis2017].
+"""
+function dist_min_angle_pair(p, q)
+    #compute rotations
+    q211 = q[1] * p[1]'
+    q212 = q[2] * p[2]'
+
+    #compute values from Proposition 9
+    t_break1, c1, m1, Φ1 = dist_min_angle_pair_discontinuity_distance(q211)
+    t_break2, c2, m2, Φ2 = dist_min_angle_pair_discontinuity_distance(q212)
+
+    #check if cost is constant
+    tolMZero = 1e-15
```

Comment:
Could we calculate tolerances in this function from `eps(eltype(q))`? 1e-15 is on the order of `eps(eltype(q))` and 1e-8 is about `sqrt(eps(eltype(q)))`.

Reply:
Similarly in `dist_min_angle_pair_df_newton`.

Reply:
Sure, that's a good idea.